apps/site/lib/site_web/views/page_view.ex

```diff
@@ -10,6 +10,143 @@ defmodule SiteWeb.PageView do
 
   use SiteWeb, :view
 
+  @spec sort_string_number(String.t(), String.t()) :: boolean
+  defp sort_string_number(a, b) do
+    case {Integer.parse(a), Integer.parse(b)} do
+      {:error, _} -> a <= b
+      {_, :error} -> a <= b
+      {a_int, b_int} -> a_int <= b_int
+    end
+  end
+
+  @spec get_route(Routes.Route.id_t()) :: Routes.Route.t() | nil
+  defp get_route(id) do
+    case id do
+      "Green" -> Routes.Repo.green_line()
+      _ -> Routes.Repo.get(id)
+    end
+  end
+
+  @spec alerts([Alerts.Alert.t()]) :: Phoenix.HTML.Safe.t()
+  def alerts(alerts) do
+    routes_with_high_priority_alerts_by_mode =
+      alerts
+      |> Enum.filter(&(Alerts.Priority.priority(&1) == :high))
+      |> Enum.reduce(MapSet.new(), fn alert, routes ->
+        MapSet.union(routes, Alerts.Alert.get_entity(alert, :route))
+      end)
+      |> Enum.map(&get_route/1)
+      |> Enum.filter(& &1)
+      |> Enum.group_by(&Routes.Route.type_atom(&1.type))
+      |> (&Map.merge(%{bus: [], subway: [], ferry: [], commuter_rail: []}, &1)).()
+      |> Enum.map(fn {k, routes} ->
```

Comment:
this map/sort-by can probably be simplified by moving the matching of the key type into the a function match vs the mid-pipe case statement. Something like:

```elixir
|> Enum.sort_by(& &1.name, &sort_by_route/2)

defp sort_by_route({:bus, _}), do: &sort_string_number/2
defp sort_by_route(_), do: &<=/2
```